src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,271 @@
+import ZipScriptLoader from './zip-script-loader';
+
+describe('ZipPaymentStrategy', () => {
+
+    let orderActionCreator: OrderActionCreator;
+    let paymentActionCreator: PaymentActionCreator;
+    let paymentMethodActionCreator: PaymentMethodActionCreator;
+    let paymentMethodMock: PaymentMethod;
+    let requestSender: RequestSender;
+    let store: CheckoutStore;
+    let scriptLoader: ScriptLoader;
+    let strategy: PaymentStrategy;
+    let submitOrderAction: Observable<Action>;
+    let submitPaymentAction: Observable<Action>;
+    let zipClient: Zip;
+    let zipScriptLoader: ZipScriptLoader;
+
+    beforeEach(() => {
+        paymentMethodMock = { ...getZip() };
+        scriptLoader = createScriptLoader();
+
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethodMock);
+
+        jest.spyOn(store, 'getState');
+
+        zipClient = getZipScriptMock('approved');
+
+        zipScriptLoader = new ZipScriptLoader(scriptLoader);
+
+        jest.spyOn(zipScriptLoader, 'load')
+            .mockReturnValue(Promise.resolve(zipClient));
+        jest.spyOn(zipClient.Checkout, 'init');
+
+        requestSender = createRequestSender();
+
+        const paymentClient = createPaymentClient(store);
+        const _requestSender: PaymentMethodRequestSender = new PaymentMethodRequestSender(requestSender);
+        const paymentRequestSender = new PaymentRequestSender(createPaymentClient());
+
+        orderActionCreator = new OrderActionCreator(paymentClient, new CheckoutValidator(new CheckoutRequestSender(createRequestSender())));
+        paymentActionCreator = new PaymentActionCreator(paymentRequestSender, orderActionCreator);
+        paymentMethodActionCreator = new PaymentMethodActionCreator(_requestSender);
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        strategy = new ZipPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            zipScriptLoader
+        );
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod');
+    });
+
+    it('creates an instance of ZipPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(ZipPaymentStrategy);
```

Comment:
I don't know what you're trying to test here because you've actually constructed the object yourself. So this is always going to be true.